src/Bicep.Decompiler/ArmHelpers/ExpressionHelpers.cs

```diff
@@ -243,5 +245,39 @@ public static (string typeString, LanguageExpression nameExpression)? TryGetReso
 
             return jTokenExpression.Value.ToString().Trim('/');
         }
+
+        public static TToken ReplaceFunctionExpressions<TToken>(TToken token, Action<FunctionExpression> onFunctionExpression)
```

Comment:
>ReplaceFunctionExpressions [](start = 29, length = 26)

Btw, do we have any tests that decompile a template that uses [[ escaped expressions? (I don't see any issue - the code just reminded me to ask the question.)

Reply:
Good point, I've added some test cases for this to `edgecases/main.json`